kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -97,34 +101,29 @@ class BatchJobSubmission(session: KyuubiBatchSessionImpl, batchRequest: BatchReq
     try {
       info(s"Submitting ${batchRequest.batchType} batch job: $builder")
       val process = builder.start
-      while (appIdAndUrl.isEmpty) {
-        try {
-          builder match {
-            case sparkBatchProcessBuilder: SparkBatchProcessBuilder =>
-              sparkBatchProcessBuilder.getApplicationIdAndUrl() match {
-                case Some(appInfo) => appIdAndUrl = Some(appInfo)
-                case _ =>
-              }
-
-            case _ =>
-          }
-        } catch {
-          case e: Exception => error(s"Failed to check batch application", e)
-        }
+      var applicationStatus = currentApplicationState
+      while (applicationStatus.isEmpty) {
+        applicationStatus = currentApplicationState
         Thread.sleep(applicationCheckInterval)
       }
-      process.waitFor()
-      if (process.exitValue() != 0) {
-        throw new KyuubiException(s"Process exit with value ${process.exitValue()}")
+      val state = applicationStatus.get(ApplicationOperation.APP_STATE_KEY)
+      if (state == "KILLED" || state == "FINISHED") {
```

Comment:
it should be `if (state == "KILLED" || state == "FAILED") `?

Reply:
FAILED added